streamer/src/nonblocking/quic.rs

```diff
@@ -178,24 +179,29 @@ async fn setup_connection(
         if stake != 0 || max_unstaked_connections > 0 {
             if let Some((last_update, stream_exit)) = connection_table_l.try_add_connection(
                 &remote_addr,
-                Some(connection),
+                Some(connection.clone()),
                 timing::timestamp(),
                 max_connections_per_ip,
             ) {
+                let table_type = connection_table_l.peer_type;
                 drop(connection_table_l);
                 let stats = stats.clone();
-                let connection_table1 = connection_table.clone();
+                let connection_table = match table_type {
+                    ConnectionPeerType::Unstaked => connection_table.clone(),
+                    ConnectionPeerType::Staked => staked_connection_table.clone(),
+                };
                 tokio::spawn(handle_connection(
                     uni_streams,
                     packet_sender,
                     remote_addr,
                     last_update,
-                    connection_table1,
+                    connection_table,
                     stream_exit,
                     stats,
                     stake,
                 ));
             } else {
+                connection.close(0u32.into(), &[0u8]);
```

Comment:
This change isn't related to the PR, can you please revert? I also don't think it's necessary because when this `connection` is dropped, there shouldn't be any other references to the underlying `ConnectionRef` remaining and `quinn` should close the connection itself.